tools/CheckOutputFiles.py

```diff
@@ -0,0 +1,230 @@
+        self.absolute_tolerance = float(absolute_tolerance)
+        self.relative_tolerance = (0.0 if relative_tolerance is None else
+                                   float(relative_tolerance))
+
+    def check_h5_file(self, h5_file, test_entity, expected_entity):
+        """Perform the unit test comparisons between the dataset or group
+
+        If `expected_entity` is `None`, then the `test_entity` is compared to
+        0.0 instead (using only the absolute tolerance)
+
+        Args:
+            h5_file: An h5 File object on which to perform checks
+            test_entity: The h5 path string to the test dataset or group
+            expected_entity: The h5 path to the expected dataset or group
+        """
+
+        if isinstance(h5_file[test_entity], h5py.Dataset):
+            with self.unit_test.subTest(test_entity=test_entity,
+                                        expected_entity=expected_entity):
+                test_data = h5_file[test_entity][()]
+                if self.expected_h5_entity is not None:
+                    expected_data = h5_file[expected_entity][()]
+                    self.unit_test.assertEqual(
+                        test_data.shape, expected_data.shape,
+                        "test and expected h5 datasets must have identical"
+                        " sizes.")
+                    self.unit_test.assertEqual(
+                        test_data.dtype, expected_data.dtype,
+                        "test and expected h5 datasets must have identical"
+                        " types.")
+                    if (test_data.dtype == np.dtype(np.float) or
+                            test_data.dtype == np.dtype(np.complexfloating)):
+                        for index in np.ndindex(test_data.shape):
+                            with self.unit_test.subTest(index=index):
+                                self.unit_test.assertTrue(
+                                    np.isclose(test_data[index],
+                                               expected_data[index],
+                                               self.relative_tolerance,
+                                               self.absolute_tolerance),
+                                    msg="Compare " + str(test_data[index]) +
+                                    " to " + str(expected_data[index]))
+                    else:
+                        self.unit_test.assertEqual(test_data, expected_data)
+                else:
+                    if (test_data.dtype == np.dtype(np.floating) or
+                            test_data.dtype == np.dtype(np.complexfloating)):
+                        for index in np.ndindex(test_data.shape):
+                            with self.unit_test.subTest(index=index):
+                                self.unit_test.assertTrue(
+                                    np.isclose(test_data[index], 0.0, 0.0,
+                                               self.absolute_tolerance))
```

Comment:
I was averse to using the npt assert because the failure is a `throw`, rather than going through the unit testing framework. the subtest information should give the details of what failed -- is it not providing that information?

Reply:
Actually the unittest asserts and the numpy.testing asserts do the same thing: Both just raise an `AssertionError`. Therefore I don't think there's an advantage to using the unittest asserts. Give the numpy.testing assert a try, I think you'll like the error message :)

Reply:
Ah, I overlooked that the unittest just raises as well -- I noticed the traceback from the numpy testing and assumed something was going wrong. Thanks, I've swapped over to `allclose`.